providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/grpc.py

```diff
@@ -249,12 +266,17 @@ def _resolve(  # noqa: PLR0915
             raise GeneralError(message) from e
 
         # Got a valid flag and valid type. Return it.
-        return FlagResolutionDetails(
+        result = FlagResolutionDetails(
             value=value,
             reason=response.reason,
             variant=response.variant,
         )
 
+        if response.reason == Reason.STATIC and self._cache is not None:
+            self._cache.insert(flag_key, result)
```

Comment:
We need to only use the cache if the stream connection is active; and flush it when the stream disconnects.

Reply:
See: https://flagd.dev/reference/specifications/providers/#cache-invalidation

Reply:
done, each time we generate a new stub and channel i will clear the cache